lang/java/reef-bridge-client/src/main/avro/BootstrapArgs.avsc

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+[
+  {
+      "namespace": "org.apache.reef.reef.bridge.client",
+      "type": "record",
+      "name": "AvroBootstrapArgs",
+      "fields": [
+        { "name": "jobId", "type": "string" },
+        { "name": "tcpBeginPort", "type": "int" },
+        { "name": "tcpRangeCount", "type": "int" },
+        { "name": "tcpTryCount", "type": "int" },
+        { "name": "jobSubmissionFolder", "type": "string" }
+      ]
+  },
+  {
+      "namespace": "org.apache.reef.reef.bridge.client",
+      "type": "record",
+      "name": "AvroLocalBootstrapArgs",
+      "fields": [
+        { "name": "sharedBootstrapArgs", "type": "AvroBootstrapArgs" },
+        { "name": "numberOfEvaluators", "type": "int" }
+      ]
+  },
+  {
+      "namespace": "org.apache.reef.reef.bridge.client",
+      "type": "record",
+      "name": "AvroYarnBootstrapArgs",
+      "fields": [
+        { "name": "sharedBootstrapArgs", "type": "AvroBootstrapArgs" },
+        { "name": "driverMemory", "type": "int" },
+        { "name": "driverRecoveryTimeout", "type": "int" },
+        { "name": "jobSubmissionDirectoryPrefix", "type": "string" }
+      ]
+  },
+  {
+      "namespace": "org.apache.reef.reef.bridge.client",
+      "type": "record",
+      "name": "AvroYarnClusterBootstrapArgs",
```

Comment:
This is because submission on the cluster through Yarn's own submission client requires a few more parameters than "direct" submission via the REST API